iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,319 @@
+    PoshRuntime* m_senderRuntime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::CString100 m_runnableName{"testRunnable"};
+    const iox::cxx::CString100 m_invalidRunnableName{"invalidRunnable,"};
+    static bool m_errorHandlerCalled;
+};
+
+bool PoshRuntime_test::m_errorHandlerCalled{false};
+
+
+TEST_F(PoshRuntime_test, ValidAppname)
+{
+    std::string appName("/valid_name");
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(appName); });
+}
+
+
+TEST_F(PoshRuntime_test, AppnameLength_OutOfLimit)
+{
+    std::string tooLongName(100, 's');
+    tooLongName.insert(0, 1, '/');
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(tooLongName); },
+                 "Application name has more than 100 characters, including null termination!");
+}
+
+
+TEST_F(PoshRuntime_test, MaxAppnameLength)
+{
+    std::string maxValidName(99, 's');
+    maxValidName.insert(0, 1, '/');
+
+    EXPECT_NO_FATAL_FAILURE({ PoshRuntime::getInstance(maxValidName); });
+}
+
+
+TEST_F(PoshRuntime_test, NoAppname)
+{
+    const std::string wrong("");
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name must not be empty!");
+}
+
+
+TEST_F(PoshRuntime_test, NoLeadingSlash_Appname)
+{
+    const std::string wrong = "wrongname";
+
+    EXPECT_DEATH({ PoshRuntime::getInstance(wrong); },
+                 "Cannot initialize runtime. Application name wrongname does not have the required leading slash '/'");
+}
+
+
+// test class creates instance of Poshruntime, so when getInstance() is called without name it reuturns extisting
+// instance
+TEST_F(PoshRuntime_test, DISABLED_AppnameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceName_ReturnValue)
+{
+    const std::string appname = "/app";
+
+    auto& sut = PoshRuntime::getInstance(appname);
+
+    EXPECT_EQ(sut.getInstanceName(), appname);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ReturnValue)
+{
+    uint32_t uniqueIdCounter = iox::popo::BasePortData::s_uniqueIdCounter;
+
+    const auto applicationPortData = m_senderRuntime->getMiddlewareApplication();
+
+    EXPECT_EQ(std::string("/sender"), applicationPortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), applicationPortData->m_serviceDescription);
+    EXPECT_EQ(false, applicationPortData->m_toBeDestroyed);
+    EXPECT_EQ(uniqueIdCounter, applicationPortData->m_uniqueId);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ApplicationlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (auto i = 0u; i < iox::MAX_PROCESS_NUMBER; ++i)
+    {
+        m_senderRuntime->getMiddlewareApplication();
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
```

Comment:
instead of just checking if the error handler is called, we should check if it's only once called and also if it was called with the expected error. So it might make sense to instantiate the `TemporaryErrorHandler` in the test case itself instead of the test fixture

Reply:
done